src/components/BlogForm/NewCard.tsx

```diff
@@ -0,0 +1,177 @@
+"use client"; // needed for useform
+import { postBlog, storageBlog } from "api/blog";
+import {
+  Button,
+  Stack,
+  TextField,
+  Box,
+  Container,
+  MenuItem,
+} from "@mui/material";
+import { useSession } from "next-auth/react";
+import { useSnackbar } from "notistack";
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { LoadingScreen } from "../global-components";
+import { DataProps } from "./types";
+
+const NewCard = () => {
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: session } = useSession();
+
+  const { control, handleSubmit, reset } = useForm({
+    defaultValues: {
+      title: "",
+      desc: "",
+      tags: "",
+    },
+  });
+
+  const [imageUpload, setImageUpload] = useState<File | null>(null);
+
+  if (!session?.user) {
+    return <LoadingScreen />;
+  }
+  const { user } = session;
+
+  const onSubmit = (data: DataProps) => {
+    if (!imageUpload) {
+      return enqueueSnackbar("No file selected.", { variant: "error" });
+    }
+    const imageId = storageBlog(imageUpload)
+      .then(() => enqueueSnackbar("Image Uploaded"))
+      .catch(() =>
+        enqueueSnackbar("Image Upload Failed", {
+          variant: "error",
+        })
+      )
+      .finally(() => setImageUpload(null));
+
+    postBlog(null, {
+      ...data,
+      author: user.name,
+      date: new Date().toDateString(),
+      authorId: user.id,
+      imageId: imageId,
+    })
+      .then(() => enqueueSnackbar("Post has been added"))
```

Comment:
Be consistent about naming, use Blog instead of Post (or say Blog post or something)